frontend/src/lib/v2/WorkflowUtils.ts

```diff
@@ -111,12 +117,19 @@ export function isPipelineSpec(templateString: string) {
 export function getContainer(componentSpec: ComponentSpec, templateString: string) {
   const executionLabel = componentSpec?.executorLabel;
 
-  const jsonTemplate = getPipelineDefFromYaml(templateString);
-  const deploymentSpec = jsonTemplate['deploymentSpec'];
+  const pipelineSpecYAML = getPipelineDefFromYaml(templateString);
+  const pipelineSpec = PipelineSpec.fromJSON(pipelineSpecYAML);
+  const deploymentSpecObj = pipelineSpec.deploymentSpec;
+  if (!pipelineSpec || !deploymentSpecObj) {
+    return null;
+  }
 
-  const executorsMap = deploymentSpec['executors'];
-  if (!executorsMap || !executionLabel) {
+  const deploymentSpec = PipelineDeploymentConfig.fromJSON(deploymentSpecObj);
+  const executorsObj = deploymentSpec.executors;
+  if (!executorsObj || !executionLabel) {
```

Comment:
It is probably better to check `executionLabel` earlier in this function? (like line 119). If `executionLabel` is empty, we can skip all the extraction of pipeline template.